### PR TITLE
Nc feature locker modal

### PR DESCRIPTION
**Pull request by nancychuchu**
Add button in findLocker page which opens modal for Locker Info which displays reviews data for a locker.

### CODE REVIEW

File: client/js/assets/icons/actions/floatingbackbutton.svg [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (siddharthparmar7) [inline client/js/assets/icons/actions/floatingbackbutton.svg]: Please use pngs. Rendering svgs can be a pain in react native.

File: client/js/assets/icons/addspotv3.svg [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (siddharthparmar7) [inline client/js/assets/icons/addspotv3.svg]: Same as above.

File: client/js/assets/icons/appheader/hamburgermenu.svg [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (siddharthparmar7) [inline client/js/assets/icons/appheader/hamburgermenu.svg]: Same as above.

File: client/js/assets/icons/appheader/location.svg [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (siddharthparmar7) [inline client/js/assets/icons/appheader/location.svg]: Same as above.

File: client/js/assets/icons/bikelockerlogo.svg [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (siddharthparmar7) [inline client/js/assets/icons/bikelockerlogo.svg]: Same as above.

File: client/js/assets/icons/bikerack.svg [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (siddharthparmar7) [inline client/js/assets/icons/bikerack.svg]: Same as above.

File: client/js/assets/icons/caution.svg [hunk 7]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (siddharthparmar7) [inline client/js/assets/icons/caution.svg]: Same as above.

File: client/js/assets/icons/chevrons/down.svg [hunk 8]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (siddharthparmar7) [inline client/js/assets/icons/chevrons/down.svg]: Same as above.

File: client/js/assets/icons/chevrons/left2.svg [hunk 9]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (siddharthparmar7) [inline client/js/assets/icons/chevrons/left2.svg]: Same as above.

File: client/js/assets/icons/chevrons/right2.svg [hunk 10]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (siddharthparmar7) [inline client/js/assets/icons/chevrons/right2.svg]: Same as above.

File: client/js/assets/icons/chevrons/up.svg [hunk 11]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (siddharthparmar7) [inline client/js/assets/icons/chevrons/up.svg]: Same as above.

File: client/js/assets/icons/emptyavatar.svg [hunk 12]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (siddharthparmar7) [inline client/js/assets/icons/emptyavatar.svg]: Same as above.

File: client/js/assets/icons/emptybikeavatar.svg [hunk 13]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (siddharthparmar7) [inline client/js/assets/icons/emptybikeavatar.svg]: Same as above.

File: client/js/assets/icons/fillableavatar.svg [hunk 14]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (siddharthparmar7) [inline client/js/assets/icons/fillableavatar.svg]: Same as above.

File: client/js/assets/icons/pins/0.svg [hunk 15]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (siddharthparmar7) [inline client/js/assets/icons/pins/0.svg]: Same as above.

File: client/js/assets/icons/pins/1.svg [hunk 16]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (siddharthparmar7) [inline client/js/assets/icons/pins/1.svg]: Same as above.

File: client/js/assets/icons/pins/1.svg [hunk 17]
```diff
@@ -0,0 +1,10 @@
...
+<svg width="44px" height="66px" viewBox="0 0 44 66" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
```
Comment (siddharthparmar7) [inline client/js/assets/icons/pins/1.svg]: Also please name your files more descriptive. 1 is not a file name.

File: client/js/assets/icons/pins/2.svg [hunk 18]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (siddharthparmar7) [inline client/js/assets/icons/pins/2.svg]: Same as above.

File: client/js/assets/icons/pins/3.svg [hunk 19]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (siddharthparmar7) [inline client/js/assets/icons/pins/3.svg]: Same as above.

File: client/js/assets/icons/pins/4.svg [hunk 20]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (siddharthparmar7) [inline client/js/assets/icons/pins/4.svg]: Same as above.

File: client/js/assets/icons/pins/5.svg [hunk 21]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (siddharthparmar7) [inline client/js/assets/icons/pins/5.svg]: Same as above.

File: client/js/assets/icons/pins/Free.svg [hunk 22]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (siddharthparmar7) [inline client/js/assets/icons/pins/Free.svg]: Same as above.

File: client/js/assets/icons/pins/Paid.svg [hunk 23]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (siddharthparmar7) [inline client/js/assets/icons/pins/Paid.svg]: Same as above.

File: client/js/assets/icons/pins/Translink.svg [hunk 24]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (siddharthparmar7) [inline client/js/assets/icons/pins/Translink.svg]: Same as above.

File: client/js/components/Loader/Loader.js [hunk 25]
```diff
@@ -0,0 +1,12 @@
+import React from "react";
+import { Text, View } from "react-native";
+
+const Loader = () => {
+  return (
+    <View>
+      <Text>This is the loader screen</Text>
```
Comment (siddharthparmar7) [inline client/js/components/Loader/Loader.js]: Please use the Activity Indicator and remove the View component.

File: client/js/components/PhotoCarousel/PhotoCarousel.js [hunk 26]
```diff
@@ -0,0 +1,63 @@
...
+            showsHorizontalScrollIndicator={false}
+          >
+            {images.map((image, index) => (
+              <Image key={index} style={styles.image} source={image.source} />
+            ))}
+          </ScrollView>
+        </View>
+      );
+    }
+    console.log("Please provide images");
```
Comment (siddharthparmar7) [inline client/js/components/PhotoCarousel/PhotoCarousel.js]: Please remove this console log.

File: client/js/components/PhotoCarousel/PhotoCarousel.js [hunk 27]
```diff
@@ -0,0 +1,63 @@
...
+    return null;
+  }
+}
+
+export default class PhotoCarousel extends Component {
+  render() {
+    const images = [
```
Comment (siddharthparmar7) [inline client/js/components/PhotoCarousel/PhotoCarousel.js:29]: put this const outside render if possible. If there is a huge array in a render, every time you render you are creating a new const with the same information.

File: client/js/components/PhotoCarousel/styles.js [hunk 28]
```diff
@@ -0,0 +1,22 @@
...
+  scrollContainer: {
+    height
+  },
+  image: {
+    width,
+    height,
+    paddingRight: 3
+  }
+});
+
```
Comment (siddharthparmar7) [inline client/js/components/PhotoCarousel/styles.js:21]: Nice work!

File: client/js/navigation/RootStackNavigator.js [hunk 29]
```diff
@@ -1,10 +1,21 @@
...
+      Locker: { screen: LockerModal, gestureResponseDistance: 50 }
+    },
+    {
+      mode: "modal",
+      headerMode: "none",
+      transparentCard: true,
+      defaultNavigationOptions: {
+        gesturesEnabled: true
+      }
+    }
```
Comment (siddharthparmar7) [inline client/js/navigation/RootStackNavigator.js:20]: Nice work!

File: client/js/screens/FindLocker/FindLocker.js [hunk 30]
```diff
@@ -1,10 +1,19 @@
...
+        navigation.push("Locker");
+      }}
+      title="Locker Info"
+      color="#841584"
+    />
   </View>
 );
 
-export default FindLocker;
+export default withNavigation(FindLocker);
```
Comment (siddharthparmar7) [inline client/js/screens/FindLocker/FindLocker.js]: use navigation as a prop (pass it from the container), not from withNavigator, for the screens.

File: client/js/screens/Locker/Locker.js [hunk 31]
```diff
@@ -1,10 +1,96 @@
...
+  ScrollView
+} from "react-native";
+import styles from "./styles";
+import PhotoCarousel from "../../components/PhotoCarousel";
+// import getDirections from "react-native-google-maps-directions";
+// import { OpenMapDirections } from "react-native-navigation-directions";
+import { Gravatar, GravatarApi } from "react-native-gravatar";
+
+class Locker extends Component {
+  constructor(props) {
```
Comment (siddharthparmar7) [inline client/js/screens/Locker/Locker.js:17]: Please remove commented code.

File: client/js/screens/Locker/Locker.js [hunk 32]
```diff
@@ -1,10 +1,96 @@
...
+    super(props);
+    this.state = {};
+  }
```
Comment (siddharthparmar7) [inline client/js/screens/Locker/Locker.js:25]: If you do not have anything in your state, remove the constructor. Thank you.

File: client/js/screens/Locker/Locker.js [hunk 33]
```diff
@@ -1,10 +1,96 @@
...
+                <Gravatar
+                  options={{
+                    email: entry.reviewer.email,
+                    parameters: { size: "200", d: "mm" },
+                    secure: true
+                  }}
+                  style={styles.roundedProfileImage}
+                />
+                <View style={styles.review}>
+                  {/* implement handling if they didn't fill first and lastname */}
```
Comment (siddharthparmar7) [inline client/js/screens/Locker/Locker.js]: Please remove commented code.

File: client/js/screens/Locker/LockerContainer.js [hunk 34]
```diff
@@ -8,6 +11,44 @@ export default class LockerContainer extends Component {
   }
```
Comment (siddharthparmar7) [inline client/js/screens/Locker/LockerContainer.js]: Remove your constructor if you don't have a state.

File: client/js/screens/Locker/styles.js [hunk 35]
```diff
@@ -0,0 +1,91 @@
...
+  container: {
+    flex: 1
+  },
+  infoContainer: {
+    flex: 1,
+    margin: 15
+  },
+  icon: {
+    width: 25,
+    height: 25
```
Comment (siddharthparmar7) [inline client/js/screens/Locker/styles.js]: can you use dynamic height and width?

File: client/js/screens/Locker/styles.js [hunk 36]
```diff
@@ -0,0 +1,91 @@
...
+    width: "100%",
+    justifyContent: "space-between",
+    paddingTop: 10,
+    paddingBottom: 15
+  },
+  roundedProfileImage: {
+    width: 50,
+    height: 50,
+    borderRadius: 25,
+    marginRight: 15
```
Comment (siddharthparmar7) [inline client/js/screens/Locker/styles.js]: can you use dynamic height and width? use resizeMode property with images.

File: client/js/screens/Locker/styles.js [hunk 37]
```diff
@@ -0,0 +1,91 @@
...
+  },
+  reviewer: {
+    flexDirection: "row",
+    paddingTop: 10,
+    paddingBottom: 10
+  },
+  //Text
```
Comment (siddharthparmar7) [inline client/js/screens/Locker/styles.js]: remove commented code.

File: client/js/screens/Locker/styles.js [hunk 38]
```diff
@@ -0,0 +1,91 @@
...
+    fontFamily: globalStyles.titleFont,
+    fontSize: 15
+  },
+  time: {
+    fontFamily: globalStyles.subheadingFont,
+    fontSize: 10,
+    marginBottom: 4
+  },
+  review: {
+    // fontFamily: globalStyles.bodyFont,
```
Comment (siddharthparmar7) [inline client/js/screens/Locker/styles.js]: Please remove commented code.

**Review by siddharthparmar7 — CHANGES_REQUESTED** [review]
Please make the changes. Thank you.

File: client/js/screens/Locker/styles.js [hunk 39]
```diff
@@ -0,0 +1,100 @@
+import { StyleSheet, Dimensions } from "react-native";
+import globalStyles from "../../config/globalStyles.js";
+
+const { width } = Dimensions.get("window");
```
Comment (teelau) [inline client/js/screens/Locker/styles.js:4]: `height and `width` are keywords, rename them for example to screenWidth.

File: client/js/screens/Locker/Locker.js [hunk 40]
```diff
@@ -1,10 +1,109 @@
...
+    };
+  }
+  setModalVisible(visible) {
+    this.setState({ modalVisible: visible });
+  }
+  render() {
+    const { lockerinfo } = this.props;
+
+    const srcLatitude = 49.2633479;
+    const srcLongitude = -123.140316;
```
Comment (teelau) [inline client/js/screens/Locker/Locker.js:30]: This can be read from the device API instead of being hardcoded.

**Review by teelau — CHANGES_REQUESTED** [review]

File: client/js/screens/Locker/Locker.js (hunk 40 above)
Comment (nancychuchu) [reply client/js/screens/Locker/Locker.js:30]: They're dummy data for now while I wait for Jeremiah's map issues to be merged.

**Review by teelau — DISMISSED** [review]
Fix merge conflicts

File: client/js/components/PhotoCarousel/PhotoCarousel.js (hunk 26 above)
Comment (siddharthparmar7) [reply client/js/components/PhotoCarousel/PhotoCarousel.js]: Please resolve this issue.

File: client/js/screens/Locker/Locker.js (hunk 31 above)
Comment (siddharthparmar7) [reply client/js/screens/Locker/Locker.js]: Please resolve this issue.

File: client/js/screens/Locker/Locker.js (hunk 32 above)
Comment (siddharthparmar7) [reply client/js/screens/Locker/Locker.js]: Please resolve this issue.

File: client/js/screens/Locker/Locker.js (hunk 33 above)
Comment (siddharthparmar7) [reply client/js/screens/Locker/Locker.js]: Please resolve this issue.

File: client/js/screens/FindLocker/FindLocker.js (hunk 30 above)
Comment (siddharthparmar7) [reply client/js/screens/FindLocker/FindLocker.js]: Please resolve this issue.

File: client/js/screens/Locker/Locker.js [hunk 41]
```diff
@@ -1,10 +1,109 @@
...
+                  <Gravatar
+                    options={{
+                      email: entry.reviewer.email,
+                      parameters: { size: "200", d: "mm" },
+                      secure: true
+                    }}
+                    style={styles.roundedProfileImage}
+                  />
+                  <View style={styles.review}>
+                    {/* implement handling if they didn't fill first and lastname */}
```
Comment (siddharthparmar7) [inline client/js/screens/Locker/Locker.js]: Please remove this comment if not needed anymore.

**Review by siddharthparmar7 — CHANGES_REQUESTED** [review]
Please make the changes and mark them resolved. Thank you.

**Review by teelau — DISMISSED** [review]

File: client/js/components/PhotoCarousel/PhotoCarousel.js (hunk 26 above)
Comment (siddharthparmar7) [reply client/js/components/PhotoCarousel/PhotoCarousel.js]: ????

File: client/js/components/Loader/Loader.js (hunk 25 above)
Comment (siddharthparmar7) [reply client/js/components/Loader/Loader.js]: ??

File: client/js/components/GoogleMapSearch/GoogleMapsSearch.js [hunk 42]
```diff
@@ -18,7 +18,7 @@ const GooglePlacesInput = () => {
         fetchDetails={true}
         renderDescription={row => row.description}
         onPress={(data, details = null) => {
-          this.props.updateContextLocation(details.geometry.location);
+          updateContextLocation(details.geometry.location);
           console.log(details);
```
Comment (siddharthparmar7) [inline client/js/components/GoogleMapSearch/GoogleMapsSearch.js]: Please remove this console log.

**Review by siddharthparmar7 — CHANGES_REQUESTED** [review]
Please resolve all the issues and use an Activity Indicator for your Loader component. Thank you.

**Review by siddharthparmar7 — APPROVED** [review]

**Pull request merged**